src/scportrait/tools/ml/datasets.py

```diff
@@ -212,382 +322,240 @@ def stats(self):  # print dataset statistics
-    def stats(self):
-        targets = [
-            info[0] for info in self.data_locator
-        ]  # get all targets from data locator
-        targets = np.array(targets, dtype=float)  # convert to numpy array
-
-        print(f"Total samples: {len(targets)}")
-
-    def __len__(self):
-        return len(self.data_locator)  # return length of data locator
-
-    def __getitem__(self, idx):
-        if torch.is_tensor(idx):
-            idx = idx.tolist()  # convert tensor to list
-
-        data_item = self.data_locator[
-            idx
-        ]  # get the data info for the current index, such as target, handle id, and row
-
-        if self.select_channel is not None:  # select a specific channel
-            cell_tensor = self.handle_list[data_item[1]][
-                data_item[2], self.select_channel
-            ]
-            t = torch.from_numpy(cell_tensor).float()  # convert to float tensor
-            t = torch.unsqueeze(t, 0)  # add channel dimension to tensor
-        else:
-            cell_tensor = self.handle_list[data_item[1]][data_item[2]]
-            t = torch.from_numpy(cell_tensor).float()  # convert to float tensor
-
-        if self.transform:
-            t = self.transform(t)  # apply transformation to the data
 
-        target = torch.tensor(data_item[0], dtype=torch.float)  # get target value
+class LabelledHDF5SingleCellDataset(_HDF5SingleCellDataset):
+    """
+    Class for handling scPortrait single cell datasets stored in HDF5 files.
+    Single-cell labels are read directly from the HDF5 file.
 
-        if self.return_id:
-            ids = int(data_item[3])
-            sample = (t, target, torch.tensor(ids))  # return data, target, and id
-        elif self.return_fake_id:
-            sample = (t, target, torch.tensor(0))  # return data, target, and fake id
-        else:
-            sample = (t, target)  # return data and target
+    This class provides a convenient interface for scPortrait formatted hdf5 files containing single cell datasets. It supports loading data
+    from multiple hdf5 files within specified directories, applying transformations on the data, and returning
+    the required information, such as label or id, along with the single cell data.
 
-        return sample
+    Attributes
+    ----------
+    dir_list : list of str
+        List of path(s) where the hdf5 files are stored. Supports specifying a path to a specific hdf5 file or directory
+        containing hdf5 files.
+    label_colum: int
+        index of column from single_cell_index_labelled from which single-cell labels should be read
+    label_dtype: dtype | None
+        dtype to which the read labels should be converted
+    label_column_transform: function| None
+        optional function that can define a mathematical transform on the read labels. E.g. if the labels are saved as seconds
+        in the HDF5 dataset you can set this value to `lambda x: x/3600` to have the labels returned in hours instead.
```

Comment:
I really like the idea of passing a transform to the labels, but i feel like giving functions directly might get complicated. Maybe limit transforms to a list with some easy examples

Reply:
True. At the moment I am not sure of the different use cases though. We had the one with transforming seconds to hours but I haven't come across any others. I think I would leave it like this for the moment because it will ensure flexibility. I also think we will still be changing quite a few things here in the future.